### DISCUSSION TITLE
Neither Cryptography 38.0.4 or Cryptography 40.0.1 works

### DISCUSSION THREAD

**Discussion by david-dong828** [aws/aws-cli#8673] [Q&A]
### Describe the bug

-------------------------------------------
cryptography 38.0.4 is installed 
Name: cryptography
Version: 38.0.4
Summary: cryptography is a package which provides cryptographic recipes and primitives to Python developers.
Home-page: https://github.com/pyca/cryptography
Author: The Python Cryptographic Authority and individual contributors
Author-email: cryptography-dev@python.org
License: BSD-3-Clause OR Apache-2.0
Location: /home/ec2-user/.local/lib/python3.9/site-packages
Requires: cffi
Required-by: acme, awscli, certbot, josepy, pyOpenSSL
--------------------------------------------

Trying to copy file from S3 to EC2.
Use command: 
[ec2-user@ip-xx-xx]$ aws s3 cp s3://ec2-1-s3/aws_test.py test1.py

### Expected Behavior

The file shoube be copied from S3 to EC2.

### Current Behavior

Error:
No module named 'cryptography'

### Reproduction Steps

 EC2: Amazon Linux 2023, US-EAST-1
S3: US-EAST-1

Then normal copy file command from S3 to EC2

### Additional Information/Context

![1706720648(1)](https://github.com/aws/aws-cli/assets/106771290/e1c3d531-f25a-4d57-9a5c-bda28c9af989)

### CLI version used

awscli-2-2.9.19-1.amzn2023.0.1.noarch

### Environment details (OS name and version, etc.)

 EC2: Amazon Linux 2023, US-EAST-1 S3: US-EAST-1

**Comment by tony12tl** [comment]
Hi David,

Try with PYTHONPATH setting as following. It worked for me. Don't know how to explain it yet.

`export PYTHONPATH=/usr/local/lib64/python3.9/site-packages`

My clue came from the location of `cryptography` module.

<img width="963" alt="Screenshot 2024-02-08 at 5 15 58 pm" src="https://github.com/aws/aws-cli/assets/112038776/9b8cd5a6-624e-43fb-8584-63dff0428747">

**Comment by tim-finnigan** [comment]
Thanks for reaching out and for your patience. Cryptography is a required dependency and [here are the supported version ranges](https://github.com/aws/aws-cli/blob/e31612ef0c01959bf9ed689128a64043871d402d/pyproject.toml#L33) for v2. I recommend uninstalling and reinstalling the CLI following [this documentation](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html) to update to a more recent version and ensure you have the required dependencies. The latest CLI version per the [CHANGELOG](https://github.com/aws/aws-cli/blob/v2/CHANGELOG.rst?plain=1) is 2.15.45. Converting this to a discussion as this is a guidance-related question, let us know if you have any follow up questions.
